controller/documentcontroller.php

```diff
@@ -50,19 +54,29 @@ class DocumentController extends Controller {
 	private $appConfig;
 	private $cache;
 	private $logger;
+	private $storage;
 	const ODT_TEMPLATE_PATH = '/assets/odttemplate.odt';
 
 	// Signifies LOOL that document has been changed externally in this storage
 	const LOOL_STATUS_DOC_CHANGED = 1010;
 
-	public function __construct($appName, IRequest $request, IConfig $settings, AppConfig $appConfig, IL10N $l10n, $uid, ICacheFactory $cache, ILogger $logger){
+	public function __construct($appName,
+								IRequest $request,
+								IConfig $settings,
+								AppConfig $appConfig,
+								IL10N $l10n,
+								$uid,
```

Comment:
uid is read from the user session? never do that - inject the user session into this class ...

Reply:
Yes, this will be taken care in follow up, refactor PR.